packages/timeline-state-resolver/src/service/commandExecutor.ts

```diff
@@ -30,16 +30,27 @@ export class CommandExecutor<DeviceState, Command extends CommandWithContext> {
 		commands: Command[],
 		measurement?: Measurement
 	): Promise<void> {
+		const start = Date.now() // note - would be better to use monotonic time here but BigInt's are annoying
+
 		await Promise.allSettled(
 			commands.map(async (command) => {
-				const timeToWait = totalTime - (command.preliminary ?? 0)
-				if (timeToWait > 0) await wait(totalTime)
+				const targetTime = start + totalTime - (command.preliminary ?? 0)
+
+				const timeToWait = targetTime - Date.now()
+				if (timeToWait > 0) {
+					await sleep(timeToWait)
+				}
 
 				measurement?.executeCommand(command)
-				return this.sendCommand(command).then(() => {
+				let result: unknown
```

Comment:
This `result` handling is confusing.

As far as I can tell, this returns `void`, so `result` should always contain `void`. 

Additionally, this flow makes the catch look weird, as it doesnt set `result`, which means that it if was returning something before it has become `| undefined`